orcid-web/src/main/java/org/orcid/frontend/web/controllers/HomeController.java

```diff
@@ -129,5 +139,64 @@ Object getUserStatusJson(HttpServletRequest request, HttpServletResponse respons
+    
+    @RequestMapping(value = "/userInfo.json", method = RequestMethod.GET)
+    public @ResponseBody Map<String, String> getUserInfo(HttpServletRequest request) {
+        Map<String, String> info = new HashMap<String, String>();        
+        OrcidProfileUserDetails userDetails = getCurrentUser();
+        if(userDetails != null) {
+            String effectiveOrcid = getEffectiveUserOrcid();
+            String realUserOrcid = getRealUserOrcid();
+            info.put("REAL_USER_ORCID", realUserOrcid);
+            // REAL_USER_ORCID = EFFECTIVE_USER_ORCID unless it is in delegation mode
+            info.put("EFFECTIVE_USER_ORCID", effectiveOrcid);
+            info.put("IN_DELEGATION_MODE", String.valueOf(!effectiveOrcid.equals(realUserOrcid)));
+            info.put("PRIMARY_EMAIL", userDetails.getPrimaryEmail());
+            info.put("HAS_VERIFIED_EMAIL", String.valueOf(emailManagerReadOnly.haveAnyEmailVerified(effectiveOrcid)));
+            info.put("IS_PRIMARY_EMAIL_VERIFIED", String.valueOf(emailManagerReadOnly.isPrimaryEmailVerified(effectiveOrcid)));
+            for(OrcidWebRole role : userDetails.getAuthorities()) {
+                switch (role) {
+                case ROLE_USER: 
+                    break;
+                case ROLE_ADMIN:
+                    info.put("ADMIN_MENU", String.valueOf(true));
+                    break;
+                case ROLE_GROUP:
+                case ROLE_BASIC:
+                case ROLE_PREMIUM:
+                case ROLE_BASIC_INSTITUTION:
+                case ROLE_PREMIUM_INSTITUTION:
+                    info.put("MEMBER_MENU", String.valueOf(true));
+                    break;
+                case ROLE_CREATOR:
+                case ROLE_PREMIUM_CREATOR:
+                case ROLE_UPDATER:
+                case ROLE_PREMIUM_UPDATER:
+                    info.put("CLIENT_MENU", String.valueOf(true));
+                    break;
+                case ROLE_SELF_SERVICE:
+                    info.put("SELF_SERVICE_MENU", String.valueOf(true));
+                }                
+            }
+            if(isDelegatedByAdmin()) {
+                info.put("DELEGATED_BY_ADMIN", String.valueOf(true));
+            }   
+            ProfileEntity p = profileEntityCacheManager.retrieve(effectiveOrcid);
+            info.put("LAST_MODIFIED", String.valueOf(p.getLastModified()));
+            info.put("DEVELOPER_TOOLS_ENABLED", String.valueOf(p.getEnableDeveloperTools()));
+            info.put("LOCKED", String.valueOf(!p.isAccountNonLocked()));
+            info.put("CLAIMED", String.valueOf(p.getClaimed()));
+            if(p.getPrimaryRecord() != null) {
+                info.put("PRIMARY_RECORD", p.getPrimaryRecord().getId());                
+            }
+            if(!PojoUtil.isEmpty(p.getGroupType())) {
+                info.put("MEMBER_TYPE", p.getGroupType());
+            }
+            
+        } else {
+            info.put("SIGN_IN_MENU", String.valueOf(true));
```

Comment:
@amontenegro not sure we need SIGN_IN_MENU ...we could just check if userInfo.json is empty